### PR TITLE
Update with branch and project structure

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

- Add information about our branching strategy for contributors
- Add project structure information for anyone getting started. I didn't create an exhaustive list of all the projects just what first time contributors would care about.

### CODE REVIEW

**Comment by mattleibow** [comment]
Should this be against main?

File: .github/DEVELOPMENT.md [hunk 1]
```diff
@@ -215,6 +215,76 @@ public static int foo = 2130771968;
...
 5. Restore the commented code in `Directory.Build.targets`.
 
+## What branch should I use?
+
+- net7.0
+  - I want to use the net7.0 sdk and make changes that will be released with the .NET 7 release of MAUI
+- net6.0
+  - This PR seems like it should go out with a net6.0 service release
+- main (start here if you don't know what to use)
+  - I want to use the net6.0 sdk and make changes that will be released with the .NET7 release of MAUI
```
Comment (TanayParikh) [inline .github/DEVELOPMENT.md:225]: Should this info be added to the [default PR template](https://github.com/dotnet/maui/blob/main/.github/PULL_REQUEST_TEMPLATE.md)?

Comment (Eilon) [inline .github/DEVELOPMENT.md:225]:
```suggestion
  - I want to use the net6.0 sdk and make changes that will be released with the .NET 7 release of MAUI
```

**Review by Eilon — APPROVED** [review]

**Comment by PureWeen** [comment]
> Should this be against main?

I debated about that, but it does all apply to the net6.0 branch as well so seemed like it should start its life there

**Review by jsuarezruiz — APPROVED** [review]

**Review by rmarinho — APPROVED** [review]

File: .github/DEVELOPMENT.md (hunk 1 above)
Comment (PureWeen) [reply .github/DEVELOPMENT.md]: Good thought!
https://github.com/dotnet/maui/pull/8420

**Pull request merged**